src/graphql/apollo/client.ts

```diff
@@ -1,15 +1,106 @@
 import { ApolloOfflineClient } from 'offix-client';
 import { InMemoryCache } from 'apollo-cache-inmemory';
 import { HttpLink } from 'apollo-link-http';
+import { onError } from 'apollo-link-error';
+import { ApolloLink, split } from 'apollo-link';
+import { WebSocketLink } from 'apollo-link-ws';
+import { getMainDefinition } from 'apollo-utilities';
+import { SubscriptionClient } from 'subscriptions-transport-ws';
+import gql from 'graphql-tag';
 
-const config = {
-  link: new HttpLink({
-    uri: process.env.REACT_APP_BACKEND_HOST || 'http://localhost:4000/',
-  }),
+const myClient = new SubscriptionClient(
+  process.env.REACT_APP_BACKEND_WEBSOCKET_URL || 'ws://localhost:4000/graphql',
+  {
+    reconnect: true,
+  }
+);
+
+const wsLink = new WebSocketLink(myClient);
+
+const httpLink: ApolloLink = new HttpLink({
+  uri: process.env.REACT_APP_BACKEND_HOST || 'http://localhost:4000/',
+  credentials: 'same-origin',
+});
+
+const link: ApolloLink = split(
+  // split based on operation type
+  ({ query }) => {
+    const definition = getMainDefinition(query);
+    return (
+      definition.kind === 'OperationDefinition' &&
+      definition.operation === 'subscription'
+    );
+  },
+  wsLink,
+  httpLink
+);
+
+const client = new ApolloOfflineClient({
+  link: ApolloLink.from([
+    onError(({ graphQLErrors, networkError }) => {
+      if (graphQLErrors)
+        graphQLErrors.map(({ message, locations, path }) =>
+          // eslint-disable-next-line no-console
+          console.log(
+            `[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`
+          )
+        );
+      if (networkError) {
+        // eslint-disable-next-line no-console
+        console.log(`[Network error]: ${networkError}`);
+      }
+    }),
+    link,
+  ]),
   cache: new InMemoryCache(),
-};
+});
+
+export const GET_NETWORK_STATUS = gql`
+  query updateNetworkStatus {
+    networkStatus
+  }
+`;
+
+myClient.onConnected(() => {
+  client.writeQuery({
+    query: GET_NETWORK_STATUS,
+    data: {
+      networkStatus: 'connected',
+    },
+  });
+});
+myClient.onConnecting(() => {
+  client.writeQuery({
+    query: GET_NETWORK_STATUS,
+    data: {
+      networkStatus: 'connecting...',
+    },
+  });
+});
+myClient.onDisconnected(() => {
+  client.writeQuery({
+    query: GET_NETWORK_STATUS,
+    data: {
+      networkStatus: 'disconnected',
+    },
+  });
+});
+myClient.onReconnected(() => {
+  client.writeQuery({
+    query: GET_NETWORK_STATUS,
+    data: {
+      networkStatus: 'connected',
```

Comment:
is this a typo or do we not really care about this status? @wongcmichelle you might need to audit which of these statuses we would really care about and use for offline
```suggestion
      networkStatus: 'reconnected',
```

Reply:
yeah @carelynntsai @jesskeepswimming I'm confused because I think it always says "disconnected" for me in the top right of the screen.

![image](https://user-images.githubusercontent.com/22457859/100817693-89c13e80-3416-11eb-818d-af19cca29c43.png)

Reply:
Actually I think my websocket connection is failing :/ Not sure why though. 

![image](https://user-images.githubusercontent.com/22457859/100817795-bf662780-3416-11eb-89cf-08c20c5e25fb.png)

Reply:
The issue is in the master `paramedics-web` branch, when Jason's auth PR got merged in there is no longer a web socket url to connect to so I'll have to address that first